### PR TITLE
Support for PS based on .Net Core 

### PR DESCRIPTION
**Pull request by hovsepm**
Fix for #3746

Code review comments that needs to be addressed later:
- [ ] standardize and simplify build switches in .Net  SDK and Azure Powershell projects
- [ ] Make a specific implementation for AdalToken in .Net core instead of #if defs
- [ ] Investigate how to enable Adal Tracing in Authentication in .Net Core
- [ ] Move ps1xml files from nestandrad sub-folder to main folder in the package
- [ ] in publishmodules.ps1 change string isNetCore to Boolean

### CODE REVIEW

**Comment by hovsepm** [comment]
Sign job: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-sign-netcore/

File: build.proj [hunk 1]
```diff
@@ -183,59 +184,122 @@
...
+    <CallTarget Targets="CodeSignInstaller"
+                Condition=" '$(CodeSign)' == 'true' and '$(Scope)' == 'all'" />
+    <Message Importance="high" Text="Running Static Analyser" />
     <CallTarget targets="DependencyAnalysis" ContinueOnError="ErrorAndContinue" />
-    <Exec Command="$(PowerShellCommand) -NonInteractive -NoLogo -NoProfile -Command &quot;. $(LibraryToolsFolder)\CleanupBuild.ps1 -BuildConfig $(Configuration) &quot;" />
+    <Exec Command="&quot;$(PowerShellCommand)&quot; -NonInteractive -NoLogo -NoProfile -Command &quot;. $(LibraryToolsFolder)\CleanupBuild.ps1 -BuildConfig $(Configuration) &quot;" />
   </Target>
   
+  <!-- Build .Net Core Cmdlets -->
+  <Target Name="BuildNetCore" DependsOnTargets="RestoreNugetPackages;BuildMsBuildTask">
```
Comment (twitchax) [inline build.proj]: This task does not actually depend on either of the `DependsOnTargets`, and if you remove this dependency, this task can be built from mac/linux.

Comment (twitchax) [inline build.proj]: @hovsepm, any way you can make this small change before merging?

**Comment by hovsepm** [comment]
sign job: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-sign-netcore/62/

**Comment by hovsepm** [comment]
sign job: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-sign-netcore/63/

**Comment by hovsepm** [comment]
.Net45 Sign job: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-sign/1165/

**Comment by hovsepm** [comment]
Storage Signed-off

**Comment by hovsepm** [comment]
on-demand: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/1680/

File: src/Common/Commands.Common/Properties/Resources.Designer.cs [hunk 2]
```diff
@@ -1207,11 +1208,11 @@ public class Resources {
         }
         
         /// <summary>
-        ///   Looks up a localized string similar to Internal Server Error.
+        ///   Looks up a localized string similar to public Server Error.
```
Comment (AzureRT) [inline src/Common/Commands.Common/Properties/Resources.Designer.cs]: Is this comment change intentional?  (internal server error -> public server error)
Or is this comment changed during replacing internal method to public?

File: src/Common/Commands.Common/Properties/Resources.Designer.cs [hunk 3]
```diff
@@ -1207,11 +1208,11 @@ public class Resources {
...
         /// </summary>
-        public static string InternalServerErrorMessage {
+        public static string publicServerErrorMessage {
```
Comment (AzureRT) [inline src/Common/Commands.Common/Properties/Resources.Designer.cs]: Similar to the previous comment.  Is the name change of this method intentional?

File: src/Common/Commands.Common/Properties/Resources.Designer.cs (hunk 2 above)
Comment (hovsepm) [inline src/Common/Commands.Common/Properties/Resources.Designer.cs]: @AzureRT good catch. Find and replace misuse :) I'll fix both issues

**Comment by hovsepm** [comment]
Storage Signed-off
Compute Signed-off

File: src/ResourceManager/Resources/Commands.Resources/Providers/GetAzureProviderOperationCmdlet.cs [hunk 4]
```diff
@@ -12,15 +12,24 @@
...
+using ProjectResources = Microsoft.Azure.Commands.Resources.Properties.Resources;
+#if !NETSTANDARD
+using Microsoft.Azure.Management.Resources;
+using Microsoft.Azure.Management.Resources.Models;
+#else
+using Microsoft.Azure.Management.ResourceManager;
+using Microsoft.Azure.Management.ResourceManager.Models;
+using Microsoft.Azure.Management.Authorization.Models;
+using Operation = Microsoft.Azure.Management.Authorization.Models.ProviderOperation;
+#endif
```
Comment (TianoMS) [inline src/ResourceManager/Resources/Commands.Resources/Providers/GetAzureProviderOperationCmdlet.cs]: merge the usings with the ones below?

File: src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Collections/InsensitiveDictionary.cs [hunk 5]
```diff
@@ -17,6 +17,7 @@ namespace Microsoft.Azure.Commands.ResourceManager.Cmdlets.Collections
     using Microsoft.Azure.Commands.ResourceManager.Cmdlets.Json;
     using System;
     using System.Collections.Generic;
+    using Commands.Common.Authentication.Abstractions;
```
Comment (TianoMS) [inline src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Collections/InsensitiveDictionary.cs]: > using Commands.Common.Authentication.Abstractions; [](start = 3, length = 51)

do we need this?

File: src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Implementation/Providers/GetAzureProviderCmdlet.cs [hunk 6]
```diff
@@ -12,15 +12,17 @@
 // limitations under the License.
 // ----------------------------------------------------------------------------------
 
+using ProjectResources = Microsoft.Azure.Commands.ResourceManager.Cmdlets.Properties.Resources;
```
Comment (TianoMS) [inline src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Implementation/Providers/GetAzureProviderCmdlet.cs]: merge with the ones below.

File: src/ResourceManager/Resources/Commands.Resources/Providers/GetAzureProviderOperationCmdlet.cs (hunk 4 above)
Comment (hovsepm) [inline src/ResourceManager/Resources/Commands.Resources/Providers/GetAzureProviderOperationCmdlet.cs]: @TianoMS Please clarify. I did not get what you mean.

**Comment by hovsepm** [comment]
CI - Running...
On Demand - [Running...](http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/1700/)
.Net Core Signed packages - [Running...](http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-sign-netcore/64/)
.Net 45 Signed packages - [Running...](http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-sign/1175/)


Storage **Signed-off**
Compute **Signed-off**
Network **Signed-off**
ResourceManager **Signed-off**
Websites **Signed-off**

File: src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Collections/InsensitiveDictionary.cs (hunk 5 above)
Comment (hovsepm) [inline src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Collections/InsensitiveDictionary.cs]: we should.

**Pull request merged**